src/Tomcat/TomcatModel.ts

```diff
@@ -33,6 +35,54 @@ export class TomcatModel {
         }
     }
 
+    public async updateVMOptions(serverName: string) : Promise<void> {
+        const server: TomcatServer = this.getTomcatServer(serverName);
+        const installPath: string = server.getInstallPath();
+        const catalinaBase: string = path.join(server.getStoragePath());
+        const bootStrap: string = path.join(installPath, 'bin', 'bootstrap.jar');
+        const tomcat: string = path.join(installPath, 'bin', 'tomcat-juli.jar');
+        let result: string[] = [
+            `${Constants.CLASS_PATH_KEY} ${bootStrap}${path.delimiter}${tomcat}`,
+            `${Constants.CATALINA_BASE_KEY}=${catalinaBase}`,
+            `${Constants.CATALINA_HOME_KEY}=${installPath}`
+        ];
+
+        if (!await fse.pathExists(server.vmOptionFile)) {
+            server.vmOptions = result.concat([Constants.BOOTSTRAP_FILE, '"$@"']);
+            return;
+        }
+        await new Promise((resolve: () => void): void => {
+            const lineReader: readline.ReadLine = readline.createInterface({
```

Comment:
search readline.createInterface for duplicate codes

Reply:
The other two listener are different scenarios. 

One is watch the server.xml for server configuration changes.
The other is child_process stdout.